sdks/python/apache_beam/utils/transform_service_launcher.py

```diff
@@ -0,0 +1,257 @@
+    channel_options = [("grpc.max_receive_message_length", -1),
+                       ("grpc.max_send_message_length", -1)]
+    if hasattr(grpc, 'local_channel_credentials'):
+      # TODO: update this to support secure non-local channels.
+      return grpc.secure_channel(
+          self._address,
+          grpc.local_channel_credentials(),
+          options=channel_options)
+    else:
+      return grpc.insecure_channel(self._address, options=channel_options)
+
+  def __enter__(self):
+    self.start()
+    self.wait_till_up(-1)
+
+    self._channel = self._get_channel()
+
+    from apache_beam import external
+    # with channel_factory_fn() as channel:
+    return external.ExpansionAndArtifactRetrievalStub(self._channel.__enter__())
+
+  def __exit__(self, *args):
+    self.shutdown()
+    self._channel.__exit__(*args)
+
+  def _run_docker_compose_command(self, command, output_override=None):
+    cmd = []
+    cmd.extend(self._docker_compose_start_command_prefix)
+    cmd.extend(command)
+
+    myenv = os.environ.copy()
+    myenv.update(self._environmental_variables)
+
+    process = subprocess.Popen(
+        cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, env=myenv)
+    std_out, _ = process.communicate()
+
+    if output_override:
+      output_override.write(std_out)
+    else:
+      print(std_out.decode(errors='backslashreplace'))
+
+  def start(self):
+    with self._launcher_lock:
+      self._run_docker_compose_command(['up', '-d'])
+
+  def shutdown(self):
+    with self._launcher_lock:
+      self._run_docker_compose_command(['down'])
+
+  def status(self):
+    with self._launcher_lock:
+      self._run_docker_compose_command(['ps'])
+
+  def wait_till_up(self, timeout_ms):
+    channel = self._get_channel()
+
+    # Waiting till the service is up.
+    channel_ready = grpc.channel_ready_future(channel)
+    wait_secs = .1
+    start_time = time.time()
+    while True:
+      if (time.time() - start_time) * 1000 > timeout_ms > 0:
+        raise ValueError(
+            'Transform service did not start in %s seconds.' %
+            (timeout_ms / 1000))
+      try:
+        channel_ready.result(timeout=wait_secs)
+        break
+      except (grpc.FutureTimeoutError, grpc.RpcError):
+        wait_secs *= 1.2
+        logging.log(
+            logging.WARNING if wait_secs > 1 else logging.DEBUG,
+            'Waiting for the transform service to be ready at %s.',
+            self._address)
+
+    logging.info('Transform service ' + self._project_name + ' started.')
+
+  def _get_status(self):
+    tmp = tempfile.NamedTemporaryFile(delete=False)
+    self._run_docker_compose_command(['ps'], tmp)
+    tmp.close()
+    return tmp.name
+
+
+def main(argv):
+  parser = argparse.ArgumentParser()
+  parser.add_argument('--project_name', default=None)
+  parser.add_argument('--command', default=None)
+  parser.add_argument('--port', type=int, default=-1)
+  parser.add_argument('--beam_version', default=None)
```

Comment:
We can make more use of `parser.add_argument` here instead of having if-then-raise statements later.
ie. make `--beam_version`, `--command` required; set valid choices for command ([example](https://github.com/apache/beam/blob/35efc5b0bd5b2ce93b856cdcd7f69fe1cf9c4699/playground/infrastructure/ci_cd.py#L40))

Reply:
Done.